plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/BlueOceanHook.java

```diff
@@ -0,0 +1,88 @@
+package org.jenkins.tools.test.hook;
+
+import org.jenkins.tools.test.SCMManagerFactory;
+import org.jenkins.tools.test.model.PluginCompatTesterConfig;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeCheckout;
+
+import org.apache.maven.scm.ScmFileSet;
+import org.apache.maven.scm.ScmTag;
+import org.apache.maven.scm.command.checkout.CheckOutScmResult;
+import org.apache.maven.scm.manager.ScmManager;
+import org.apache.maven.scm.repository.ScmRepository;
+import hudson.model.UpdateSite.Plugin;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Workaround for the blueocean plugins since they are
+ * stored in a central repository.
+ * 
+ */
+public class BlueOceanHook extends PluginCompatTesterHookBeforeCheckout {
+    private String parentUrl = "scm:git:git://github.com/jenkinsci/blueocean-plugin.git";
+    private String parentName = "blueocean";
+    private String parentProjectName = "blueocean-parent";
+    private List<String> allBundlePlugins = Arrays.asList("blueocean", "blueocean-commons", "blueocean-config", "blueocean-dashboard", "blueocean-events", "blueocean-git-pipeline", "blueocean-github-pipeline", "blueocean-i18n", "blueocean-jwt", "blueocean-personalization", "blueocean-pipeline-api-impl", "blueocean-rest", "blueocean-rest-impl", "blueocean-web");
+    boolean firstRun = true;
+
+    public BlueOceanHook() {}
+
+    /**
+     * All the plugins that are part of this repository.
+     */
+    public List<String> transformedPlugins() {
+        return allBundlePlugins;
+    }
+
+    /*
+     * No check implementation is required because transformedPlugins
```

Comment:
I know it's from the original MultiModule example to explain what was going on.  For the sake of file length or in the context of this particular file, it can be removed.